packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1998,15 +2004,20 @@ function commitMutationEffectsOnFiber(
               finishedWork,
               finishedWork.return,
             );
-          } finally {
-            recordLayoutEffectDuration(finishedWork);
+          } catch (error) {
+            captureCommitPhaseError(finishedWork, finishedWork.return, error);
           }
+          recordLayoutEffectDuration(finishedWork);
```

Comment:
Does it matter for the profiler whether `captureCommitPhaseError` or `recordLayoutEffectDuration` runs first? With this change, I think, their ordering changes.

Reply:
Nah don't think so. It adds a negligible amount time to the effect duration in the case where it throws, but arguably you want to account for error handling in that case.